pkgs/checks/lib/src/extensions/math.dart

```diff
@@ -0,0 +1,21 @@
+import 'package:checks/context.dart';
+
+extension NumChecks on Check<num> {
+  void isGreaterThan(num other) {
```

Comment:
If you need two `<` and `>` operations, we should give you an `.inRange(a, b)` instead. It's cleaner.

You can still do
```dart
checkThat(Box(actual)).that((v) {
  v > 0;
  v < 10;
});
```
That's one of the reasons for having `that`, as I understand it.

We can have *both*: Use `<`, `>`, `<=` and `>=` where they work, use `.greaterThan`, `.greaterThanEqual`, `lessThen` and `lessThanEqual` where they shorter versions don't work. Use `inRange` when you need to use two checks (on integers, it can be inclusive, on doubles, we might want to have both open and closed ranges).

Reply:
> we should give you an `.inRange(a, b)` instead. It's cleaner.

I think I disagree, it's either too verbose to too cryptic to specify the open/closed ranges. I think seeing the checks for each boundary is going to be more in readable. I think this becomes especially true if we switch to `>` and `>=` instead of `isGreatherThan` and `isGreaterOrEqualTo`.

> You can still do
> 
> ```dart
> checkThat(Box(actual)).that((v) {
>   v > 0;
>   v < 10;
> });
> ```

Hmm, yeah I'm not sure why I was so fixated on cascades.

I think this one could be controversial, but it's one that I suspect might get more comfortable after usage so I might give it a try in these early versions and see how it goes.

Reply:
I'm sure someone will define `.inRange` if we don't. Since it's all extension methods, anyone can add them, and they can do whichever variant they prefer - open, closed, half-open in either direction ... heck:
```dart
extension RangeCheckNum on Check<num> {
  void inRange({num? min, num? max, num? inf, num? sup}) {
    if (min != null) this >= min;
    if (max != null) this <= end;
    if (inf != null) this > inf;
    if (sup != null) this < sup;
  }
}

extension RangeCheckComparable<T extends Comparable<T>> on Check<T> {
  Check<int> compareTo(T other) => that((self) => self.compareTo(other));

  void inRange({T? min, T? max, T? inf, T? sup}) {
    if (min != null) this.compareTo(min) >= 0;
    if (max != null) this.compareTo(max) <= 0;
    if (inf != null) this.compareTo(inf) > 0;
    if (sup != null) this.compareTo(sup) < 0;
  }
}
```

and then you can just do `checkThat(string).length.inRange(min:0, max: 7)`.